### DISCUSSION TITLE
Slope/Resistance mode

### DISCUSSION THREAD

**Discussion by metaxy-nowak** [dvmarinoff/Auuki#167] [General]
I'm not sure if this is related to the Elite Suito T, but switching to Slope/Resistance mode makes riding almost impossible. The power output spikes to 500W from 100W in ERG mode. It also appears to be independent of the set value. Even setting the resistance to -10 makes it extremely difficult. Besides that, I would suggest enlarging the three buttons (ERG, Resistance, Slope) because currently, only the text is clickable, making them hard to select. Additionally, is there a way to simply enjoy a "free ride" with the wattage adjusted in-game, or must I create a basic workout with a flat intensity target?

**Comment by dvmarinoff** [comment]
Hi and welcome to the repo and thanks for the feedback!

About Resistance mode it might not be supported on the Suito, it just doesn't get implemented by most trainers. The Suito supports the common FTMS protocol so Slope mode should work.

You are right about the buttons, I just got the clickable area enlarged to match +/- buttons as size. Here is a before/after shot, where the blue color is the clickable area:

<img width="361" alt="Screenshot 2023-12-13 at 10 04 21" src="https://github.com/dvmarinoff/Flux/assets/7523832/f89a5c3d-77e7-4020-8bfc-a8de747bd398">

<img width="361" alt="Screenshot 2023-12-13 at 10 10 09" src="https://github.com/dvmarinoff/Flux/assets/7523832/0fe84170-561e-4020-909f-c8e6ab2e4002">

About free ride you have 2 options:
1. You will notice 2 play buttons at the bottom, one with circle and one without. The circled on estarts a workout, while the plain one just starts recording. If you press the plain play button you will be able to control the resistance with the +/- buttons in ERG and Slope mode.

2. You can load a route file in .fit format, and ride it in SIM mode.

**Comment by metaxy-nowak** [comment]
About slope: setting it to 0.5 spikes to 400-500W it's unusable when the slope "tab" is selected. About your solution to free ride: looks like good idea, but as someone else mentioned - icons are not particularly intuitive - shame that I don't have any idea for better ones. Two ideas:

1. Move the trainer/HRM icons to bottom row - you're using it only at the start or when connection is lost. I suggest 3 states: whole red circle, half and half, and green one.
2. Add vertical moving line to the workout graph to indicate progress in intervals.

Anyway great app, I really enjoy simple solution without making fake GPS data like in Zwift.

**Reply by dvmarinoff** [reply]
I am going to investigate the slope issue, but this same code works with Tacx , Wahoo and Zwift trainers that use the FTMS protocol and it is known to work with Elite trainers in the past. I am reluctant to change the icons, maybe  a tour guide on first use could help. About the interval time indicator vertical line my opinion is that it's redundant and would only create clutter in the UI.
